Quantum-Control-Applications/ESR/T1.py

```diff
@@ -0,0 +1,165 @@
+from qm import SimulationConfig
+from qm.qua import *
+from qm import LoopbackInterface
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+from qm.simulate.credentials import create_credentials
+
+################################
+# Open quantum machine manager #
+################################
+
+qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+
+########################
+# Open quantum machine #
+########################
+
+qm = qmm.open_qm(config)
+
+###################
+# The QUA program #
+###################
+
+pi_len = int(320//4)
+pi_half_len = int(160//4)
+
+wait_min = int(1500//4)
+wait_max = int(20000//4)
+dwait = 1000//4
+
+waits = np.arange(wait_min, wait_max + dwait/2, dwait)
+
+cooldown_time = int(10e6//4)
+
+n_avg = 100
+
+with program() as T1:
+
+    n = declare(int)
+    wait_len = declare(int)
+    pulse_delay = declare(int)
+    readout_delay = declare(int)
+
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+
+        with for_(wait_len, wait_min, wait_len <= wait_max, wait_len + dwait):
+
+            # initialization
+            # wait(wait_len)
+
+            reset_phase('ensemble')
+            reset_phase('resonator')
+            reset_frame('ensemble')
+
+            echo = declare_stream(adc_trace=True)
+
+            assign(pulse_delay, 2000//4 - Cast.mul_int_by_fixed(pi_half_len, 0.5) - Cast.mul_int_by_fixed(pi_len, 0.5))
+            assign(readout_delay, 2000//4 - Cast.mul_int_by_fixed(pi_len, 0.5) - Cast.mul_int_by_fixed(readout_len, 0.125) - 5)
+
+            wait(8, 'switch_1', 'switch_2')
```

Comment:
why wait(8)?

Reply:
because `play()` uses `duration` and is analog it takes 8 cycles to come out later in comparison to the `play` of a digital pulses that uses `duration`